Islandora/islandora
# Allow page view derived Advanced Search blocks on any page

Allows the Advanced Search block to be used on pages
where its corresponding view is not present. Now it
will redirect to the route associated with the view
it was generated from.

This is not valid for Advanced Search blocks that
are derived from non-page based views, as there is
no known location in which to redirect the query.

**GitHub Issue**: https://github.com/Islandora/documentation/issues/1869

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

https://islandora.slack.com/archives/CM5PPAV28/p1626110292349300

# What does this Pull Request do?

Allows advanced search blocks to be used on pages where the view that they are derived from is not present. Only applies to page views, as they have a location to redirect to.

# How should this be tested?

* On an existing site, place a page view derived advanced search block on say the front page.
* Confirm that searching with it does not work
* Update islandora to the latest commit in this pull request
* Clear the cache
* Attempt to search on the front page again with the advanced search block
* Should now redirect the user to the  page associated with the view used to derive the advanced search block

## Example of previous behaviour

![error](https://user-images.githubusercontent.com/487373/126348251-688b1d9d-75aa-4a0d-916a-4f63261334e1.gif)

## Behaviour after fix

![fixed](https://user-images.githubusercontent.com/487373/126348277-4004842b-a615-42b8-8bd9-e66ffd2eb8e3.gif)


# Interested parties
@Islandora/8-x-committers
